source/adapters/hip/memory.cpp

```diff
@@ -274,7 +274,12 @@ UR_APIEXPORT ur_result_t UR_APICALL urMemGetInfo(ur_mem_handle_t hMemory,
           return AllocSize;
         } else if constexpr (std::is_same_v<T, SurfaceMem>) {
           HIP_ARRAY3D_DESCRIPTOR ArrayDescriptor;
-          UR_CHECK_ERROR(hipArray3DGetDescriptor(&ArrayDescriptor, Mem.Array));
+#if HIP_VERSION >= 50600000
+          UR_CHECK_ERROR(
+              hipArray3DGetDescriptor(&ArrayDescriptor, Mem.getArray()));
+#else
+          return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
```

Comment:
I don't think this `return` is correct. We're in a lambda function that's invoked via `const auto AllocSize = std::visit(MemVisitor, hMemory->Mem);` below. Instead of returning the error code from `urMemGetInfo`, we will assign it as a value to the `AllocSize`.

Reply:
Aha good spot thanks @al42and

Reply:
Change made